consensus/state_test.go

```diff
@@ -256,73 +257,88 @@ func TestStateBadProposal(t *testing.T) {
 
-	if err := cs1.SetProposalAndBlock(proposal, propBlock, propBlockParts, "some peer"); err != nil {
-		t.Fatal(err)
-	}
+			// make the second validator the proposer by incrementing round
+			round++
+			incrementRound(vss[1:]...)
 
-	// start the machine
-	startTestRound(cs1, height, round)
+			blockID := types.BlockID{Hash: propBlock.Hash(), PartSetHeader: propBlockParts.Header()}
+			proposal := types.NewProposal(height, round, -1, blockID)
+			p := proposal.ToProto()
+			if err := vs2.SignProposal(cs1.state.ChainID, p); err != nil {
+				t.Fatal("failed to sign bad proposal", err)
+			}
+			proposal.Signature = p.Signature
 
-	t.Log("Block Sizes", "Limit", cs1.state.ConsensusParams.Block.MaxBytes, "Current", totalBytes)
+			totalBytes := 0
+			for i := 0; i < int(propBlockParts.Total()); i++ {
+				part := propBlockParts.GetPart(i)
+				totalBytes += len(part.Bytes)
+			}
 
-	// c1 should log an error with the block part message as it exceeds the consensus params. The
-	// block is not added to cs.ProposalBlock so the node timeouts.
-	ensureNewTimeout(timeoutProposeCh, height, round, cs1.config.Propose(round).Nanoseconds())
+			if err := cs1.SetProposalAndBlock(proposal, propBlock, propBlockParts, "some peer"); err != nil {
+				t.Fatal(err)
+			}
 
-	// and then should send nil prevote and precommit regardless of whether other validators prevote and
-	// precommit on it
-	ensurePrevote(voteCh, height, round)
-	validatePrevote(t, cs1, round, vss[0], nil)
+			// start the machine
+			startTestRound(cs1, height, round)
+
+			t.Log("Block Sizes;", "Limit", cs1.state.ConsensusParams.Block.MaxBytes, "Current", totalBytes)
+
+			validateHash := propBlock.Hash()
+			lockedRound := int32(1)
+			if testCase.oversized {
+				validateHash = nil
+				lockedRound = -1
+				// if the block is oversized cs1 should log an error with the block part message as it exceeds
+				// the consensus params. The block is not added to cs.ProposalBlock so the node timeouts.
+				ensureNewTimeout(timeoutProposeCh, height, round, cs1.config.Propose(round).Nanoseconds())
+				// and then should send nil prevote and precommit regardless of whether other validators prevote and
+				// precommit on it
+			}
+			ensurePrevote(voteCh, height, round)
+			validatePrevote(t, cs1, round, vss[0], validateHash)
 
-	bps, err := propBlock.MakePartSet(partSize)
-	require.NoError(t, err)
+			bps, err := propBlock.MakePartSet(partSize)
+			require.NoError(t, err)
 
-	signAddVotes(cs1, cmtproto.PrevoteType, propBlock.Hash(), bps.Header(), false, vs2)
-	ensurePrevote(voteCh, height, round)
-	ensurePrecommit(voteCh, height, round)
-	validatePrecommit(t, cs1, round, -1, vss[0], nil, nil)
+			signAddVotes(cs1, cmtproto.PrevoteType, propBlock.Hash(), bps.Header(), false, vs2)
+			ensurePrevote(voteCh, height, round)
+			ensurePrecommit(voteCh, height, round)
+			validatePrecommit(t, cs1, round, lockedRound, vss[0], validateHash, validateHash)
```

Comment:
Replaces old line 321, `lockedRound` instead of `-1` and `validateHash` instead of `nil`. Again, if `testCase.oversized`, the old values are adopted as well in the new version.